### ISSUE TITLE
More resilient Shrink Action in ILM to node dropping from cluster.

### ISSUE THREAD

**Issue by talevy** [elastic/elasticsearch#61174]
The ILM Shrink Action sets the allocation settings for the index to a specific node by `_id`. If the node with this allocation disconnects and never returns, then the index's allocation setting `index.routing.allocation.require._id` will never be valid and the shard will never be allocated correctly. There needs to be a more resilient step after the `SetSingleNodeAllocateStep` that verifies the node still exists and retries if the node is no longer part of the cluster.

**Comment by sophiaxu8** [comment]
Is this  https://github.com/elastic/elasticsearch/pull/41170 fix this issue?

**Comment by dakrone** [comment]
@sophiaxu8 no, that change was related to internal tests. We're still working on the resiliency aspect for interrupted shrink ILM operations
